### PR TITLE
modify allowed location for Linehaul Transportation(LHS) to origin: #163407346

### PR DESCRIPTION
**Pull request by lynzt**
## Description

Change Linehaul Transportation(LHS) item location from `NEITHER` to `ORIGIN`. And display `O` vs `N` where appropriate.

## Reviewer Notes

* Created a migration to update the `allowed_location`
* Database value wasn't being used when generating the base shipment line item location. Modify the code to use the database data vs hard coding the allowed location for each base shipment line item.
* Update test data to use correct location value.

## Setup

Run db migrations to ensure you have updated data.

## Code Review Verification Steps

* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](./docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/163407346) for this change

## Screenshots

TSP:
screen shot 2019-02-04 at 6 51 50 am

OFFICE:
screen shot 2019-02-04 at 6 52 08 am

### CODE REVIEW

**Comment by lynzt** [comment]
I updated `generator_test` and `update_invoice_upload_test` to ensure `LHS` has the appropriate code. This makes me wonder if we should eventually have tests to check if our data matches what we expect to see...

There were a few tests that started failing when we added validation to the `ShipmentLineItem`. Updated those tests and added a general function to display any validation error we encounter.

File: pkg/edi/invoice/generator_test.go [hunk 1]
```diff
@@ -236,13 +236,13 @@ func helperShipment(suite *InvoiceSuite) models.Shipment {
 		}
 
 		// default location created in testdatagen shipmentLineItem is DESTINATION
-		if code == "135A" || code == "105A" {
+		if code == "135A" || code == "105A" || code == "LHS" {
 			location = models.ShipmentLineItemLocationORIGIN
 		}
 		if code == "135B" {
```
Comment (reggieriser) [inline pkg/edi/invoice/generator_test.go]: I know this already existed prior to your PR, but we should be able to just do an if/else here (and the other place with the similar change).

Comment (reggieriser) [inline pkg/edi/invoice/generator_test.go]: I know this already existed prior to your PR, but we should be able to just do an if/else or a switch here (and the other place with the similar change).

File: pkg/models/shipment_line_item.go [hunk 2]
```diff
@@ -137,3 +139,27 @@ func (s *ShipmentLineItem) Approve() error {
 	s.ApprovedDate = time.Now()
 	return nil
 }
+
+// Validate gets run every time you call a "pop.Validate*" (pop.ValidateAndSave, pop.ValidateAndCreate, pop.ValidateAndUpdate) method.
+// This method is not required and may be deleted.
+func (s *ShipmentLineItem) Validate(tx *pop.Connection) (*validate.Errors, error) {
+	if s == nil {
```
Comment (reggieriser) [inline pkg/models/shipment_line_item.go]: Good catch.  TIL that you can call a method in Go using a nil receiver, assuming the receiver is a pointer and the nil is typed.  My test Go playground: https://play.golang.org/p/Pxt_LWB4I8c

File: pkg/models/shipment_line_item.go [hunk 3]
```diff
@@ -137,3 +139,27 @@ func (s *ShipmentLineItem) Approve() error {
 	s.ApprovedDate = time.Now()
 	return nil
 }
+
+// Validate gets run every time you call a "pop.Validate*" (pop.ValidateAndSave, pop.ValidateAndCreate, pop.ValidateAndUpdate) method.
```
Comment (reggieriser) [inline pkg/models/shipment_line_item.go]: Small nit, but could move this up toward the top of the file near the struct like the other models (a skeleton is normally generated by `soda`, so most models have the same rough order of these boilerplate things).

File: pkg/models/shipment_test.go [hunk 4]
```diff
@@ -233,7 +233,7 @@ func (suite *ModelSuite) TestCreateShipmentLineItem() {
 		acc.ID,
 		&q1,
 		nil,
-		"O",
+		"ORIGIN",
```
Comment (reggieriser) [inline pkg/models/shipment_test.go]: Looks like our validations already paid off by catching this!

File: pkg/models/shipment_line_item.go (hunk 3 above)
Comment (lynzt) [reply pkg/models/shipment_line_item.go]: Moved to align w/ other models. If we don't do it here, then when 😄

File: pkg/edi/invoice/generator_test.go (hunk 1 above)
Comment (lynzt) [reply pkg/edi/invoice/generator_test.go]: added `if else` to this and 2 other locations... 
* one: further down in this file
* one: in `update_invoice_upload_test.go`

**Pull request merged**